[docs] Adjust Picker field lifecycle explanation (#16901)
mui/mui-x: docs/data/date-pickers/lifecycle/lifecycle.md

It will only call the `onChange` callback when:

- the user completes all sections of a field. The value reflects the field.

>   - the value could be `Invalid Date` in case of an unparsable date (i.e., if the [year is less than "100"](https://github.com/iamkun/dayjs/issues/1237) in case of `dayjs`).

Reviewer: ```suggestion
  - the value will be `Invalid Date` in case of an unparsable date (for example, if the [year is less than "100"](https://github.com/iamkun/dayjs/issues/1237) in case of `dayjs`).
```

It will only call the `onChange` callback when:

- the user completes all sections of a field. The value reflects the field.

>   - the value will be `Invalid Date` in case of an unparsable date (for example, if the [year is less than "100"](https://github.com/iamkun/dayjs/issues/1237) in case of `dayjs`).

Reviewer: Vale...

```suggestion
  - when the date is not parseable, the `onChange` prop receives an `Invalid Date` (for example, if the [year is less than "100"](https://github.com/iamkun/dayjs/issues/1237) in case of `dayjs`).
```

If you prefer :laughing:
